### ISSUE TITLE
Temporary variables in generator expressions and listcomps lead to "name could not be sourced" error

### ISSUE THREAD

**Issue by lmmx** [lmmx/mvdef#14]
There's a "names could not be sourced" bug e.g.

```STDERR
AssertionError ⠶ These names could not be sourced: ['_']
```

Where the name is (obviously) a temporary/throwaway variable when running

```py
def foo():
    return (0 for _ in range(3))
```
⇣
```STDERR
AssertionError ⠶ These names could not be sourced: ['_']
```

**Comment by lmmx** [comment]
The names should probably be ‘screened’ in some way, and this will require classes to be instantiated during the AST parsing, exposing a string (or perhaps simply subclassed on string).

In this case, the name `_` is being erroneously required to have an imported or assigned source, but it is a temporary variable within a generator expression, similar to a list comprehension.

Similarly, this probably points to a potential flaw in `mvdef` (to be raised in a separate issue), that any reuse of a variable name is not detected (and it’s not immediately clear what the consequences of this may be).

**Comment by lmmx** [comment]
Turns out it's [already handled](https://github.com/lmmx/mvdef/blob/911ffeb9ec40e50eb139a7c1e9febabcd159fa52/src/mvdef/ast_util.py#L255-L264) for `ast.ListComp` in `ast_util.py`

```py
        elif type(a) is ast.ListComp:
            # Handle implicit assignments within list comprehensions
            for g in a.generators:
                assert type(g) is ast.comprehension, f"{a} not ast.comprehension type"
                if type(g.target) is ast.Name:
                    args_indiv.append(g.target.id)
                elif type(g.target) is ast.Tuple:
                    args_multi.extend([x.id for x in g.target.elts])
                else:
                    raise ValueError(f"{g.target} lacks expected ast.Name statements")
```

I just didn't handle the other classes

**Comment by lmmx** [comment]
Resolved in [`bcb5270`](https://github.com/lmmx/mvdef/commit/bcb52703f325f666937c784cc6191cd8ccc45a70) :+1:
